core/server/master/src/main/java/alluxio/master/file/contexts/DeleteContext.java

```diff
@@ -49,18 +49,20 @@ public static DeleteContext create(DeletePOptions.Builder optionsBuilder) {
    */
   public static DeleteContext mergeFrom(DeletePOptions.Builder optionsBuilder) {
     DeletePOptions masterOptions =
-        FileSystemOptionsUtils.deleteDefaults(Configuration.global(), false);
+        FileSystemOptionsUtils.deleteDefaults(Configuration.global(), false, false);
     DeletePOptions.Builder mergedOptionsBuilder =
         masterOptions.toBuilder().mergeFrom(optionsBuilder.build());
     return create(mergedOptionsBuilder);
   }
 
   /**
+   * @param alluxioOnly Whether to be alluxioOnly or not
    * @return the instance of {@link DeleteContext} with default values for master
    */
-  public static DeleteContext defaults() {
+  public static DeleteContext defaults(boolean alluxioOnly) {
```

Comment:
defaults should only be defaults otherwise it would be a bit confusing. Prefer to just construct your own context. It would also avoid unnecessary change impact other code path.